### PR TITLE
[RM-2076] Add aws_iam_credential_report resource

### PR DESCRIPTION
**Pull request by jaspervdj-luminal**
This is a Fugue-specific read-only resource type that just grabs all
information from the AWS IAM Credential Report.  This will allow us
to write some more CIS validations.

### CODE REVIEW

**Review by eric-luminal — APPROVED** [review]
neat that we can do this, lgtm

**Review by tyler-luminal — APPROVED** [review]

**Review by grim-luminal — COMMENTED** [review]
Are we able to use a custom diff hook to filter out any diffs for this resource?  Or does marking everything computed already do that?

File: aws/resource_aws_iam_credential_report.go [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+				AccessKey{
+					Active:       parseCsvBool(line[header["access_key_1_active"]]),
+					LastUsedDate: line[header["access_key_1_last_used_date"]],
+					LastRotated:  line[header["access_key_1_last_rotated"]],
+				},
+				AccessKey{
+					Active:       parseCsvBool(line[header["access_key_2_active"]]),
+					LastUsedDate: line[header["access_key_2_last_used_date"]],
+					LastRotated:  line[header["access_key_2_last_rotated"]],
+				},
```
Comment (grim-luminal) [inline aws/resource_aws_iam_credential_report.go:219]: Does the report include a maximum of two access keys? What if the user has more or less than two access keys defined?

File: aws/resource_aws_iam_credential_report.go [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+		// Parse report.
+		log.Printf("[INFO]: Credential Report Content: %s", string(getReportOutput.Content))
+		report, err := parseCsvCredentialReport(getReportOutput.Content)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		// Retrieve info about virtual MFA devices.
+		listMfaInput := &iam.ListVirtualMFADevicesInput{}
+		listMfaOutput, err := iamconn.ListVirtualMFADevices(listMfaInput)
```
Comment (grim-luminal) [inline aws/resource_aws_iam_credential_report.go]: I think is equivalent to `iamconn.ListVirtualMFADevices(nil)`

File: aws/resource_aws_iam_credential_report.go (hunk 1 above)
Comment (jaspervdj-luminal) [reply aws/resource_aws_iam_credential_report.go:219]: It's limited to two: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_getting-report.html#id_credentials_understanding_the_report_format

Comment (grim-luminal) [reply aws/resource_aws_iam_credential_report.go:219]: TIL

**Review by curtis-fugue — APPROVED** [review]

**Comment by curtis-fugue** [comment]
This has been looking good in Jasper's testing. No diffs due to pure computed fields.

**Pull request merged**